projects/com.oracle.truffle.llvm.parser.bc.impl/src/com/oracle/truffle/llvm/parser/bc/impl/LLVMBitcodeVisitor.java

```diff
@@ -353,4 +357,8 @@ public void visit(FunctionDefinition method) {
     @Override
     public void visit(Type type) {
     }
+
+    public LLVMTypeHelper getTypeHelper() {
```

Comment:
The `LLVMVisitor` implements `LLVMParserRuntime` to provide some common functionality to the factories. The `LLVMParserRuntime` provides some context information such as the return stack slot, stack pointer slot, bit alignment computation and so on. By using the `LLVMParserRuntime`, we do not have to extend the factories to have more and more parameters. I do not know about the `LLVMBitcodeFunctionVisitor` though.